modules/ray_tracing/include/userobjects/RayTracingStudy.h

```diff
@@ -658,9 +658,13 @@ class RayTracingStudy : public GeneralUserObject, public SidePtrHelper
   const TraceRay & traceRay(const THREAD_ID tid) const { return *_threaded_trace_ray[tid]; }
 
   /**
-   * Whether or not Ray verification is enabled in opt mode
+   * Whether or not to verify if Rays have valid information before being traced
    */
   bool verifyRays() const { return _verify_rays; }
+  /**
+   * Whether or not trace verification is enabled in devel/dbg modes
```

Comment:
A little confused, should this only be used in dbg/devel mode? Right now it seems to be controlled by an input parameter, irrelevant to compile mode.

Reply:
I could put an

```
#ifndef NDEBUG
...
#endif
```

around it. However, in test cases that run in both opt/dbg, it would be nice to just set `verify_trace_intersections = true` and have it be ignored in the opt case. I could produce a warning if it's provided in opt, but that would kill testing too.

Have any alternate ideas?

Reply:
My thought would be to put:
```
#ifndef NDEBUG
  params.addParam<bool>("verify_trace_intersections",
                        true,
                        "Whether or not to verify the trace intersections in devel and dbg modes.");
#endif
```

Reply:
I just don't particularly like having a parameter option that doesn't get used.

Reply:
The problem is that we run a single test in multiple modes (typically opt and dbg). In order to get this to work like this, you would have to create two separate tests and two requirements.

Reply:
Ah, okay. I understand your point.